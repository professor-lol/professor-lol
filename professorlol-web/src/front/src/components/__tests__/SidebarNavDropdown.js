import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import SidebarNavDropdown from "../Sidebar/SidebarNavDropdown";

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe("SidebarNavDropdown.vue", () => {
  // Inspect the raw component options
  it("should have default props", () => {
    const wrapper = shallowMount(SidebarNavDropdown, {
      localVue,
      router,
      propsData: {
        name: 'test',
        url: '',
        icon: ''
      }
    })
    expect(wrapper.props().name).toEqual('test')
    expect(wrapper.props().url).toBe('')
    expect(wrapper.props().icon).toBe('')
  });
  it('has classIcon computed property', () => {
    expect(typeof SidebarNavDropdown.computed.classIcon).toBe('function')
  })
  it('has handleClick method', () => {
    expect(typeof SidebarNavDropdown.methods.handleClick).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = shallowMount(SidebarNavDropdown, { localVue, router })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.is('router-link-stub')).toBe(true)
  })
});
