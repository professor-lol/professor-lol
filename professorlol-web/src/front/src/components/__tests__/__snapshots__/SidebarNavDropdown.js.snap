// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNavDropdown.vue renders correctly 1`] = `
<router-link-stub
  class="nav-item nav-dropdown"
  disabled=""
  event="click"
  tag="li"
  to=""
>
  <div
    class="nav-link nav-dropdown-toggle"
  >
    <i
      class="nav-icon"
    />
     
  </div>
   
  <ul
    class="nav-dropdown-items"
  />
</router-link-stub>
`;
